src/web.rs

```diff
@@ -295,6 +299,56 @@ impl ServiceInner {
         Ok(resp)
     }
 
+    fn save_camera(&self, req: &Request<::hyper::Body>, body: serde_json::Value) -> ResponseResult {
+         let (mut resp, writer) = http_serve::streaming_body(&req).build();
+        resp.headers_mut().insert(header::CONTENT_TYPE,
+                                  HeaderValue::from_static("application/json"));
+
+        let c: CameraChange = serde_json::from_value(body).map_err(|_| bad_req("missing fields"))?;
+        if c.streams.len() == 0 {
+            *resp.status_mut() = StatusCode::BAD_REQUEST;
```

Comment:
I'd do `return Err(bad_req("missing streams"))` to have a more informative error message. Same below with no such dir case.